static/js/user_profile.js

```diff
@@ -54,6 +59,120 @@ function format_user_group_list_item(group) {
+    loading.destroy_indicator($spinner);
+}
+
+function render_manage_user_tab_content(user) {
+    const $container = $("#user-profile-manage-user-tab");
+    $container.empty();
+    const user_role_values = JSON.parse(JSON.stringify(settings_config.user_role_values));
+
+    if (!page_params.is_owner && !user.is_owner) {
+        delete user_role_values.owner;
+    }
+
+    let user_email = settings_data.email_for_user_settings(user);
+    if (!user_email) {
+        // When email_address_visibility is "Nobody", we still
+        // want to show the fake email address in the edit form.
+        //
+        // We may in the future want to just hide the form field
+        // for this situation, once we display user IDs.
+        user_email = user.email;
+    }
+
+    const adminHumanForm = render_admin_human_form({
+        user_id: user.user_id,
+        email: user_email,
+        full_name: user.full_name,
+        user_role_values,
+        disable_role_dropdown: user.is_owner && !page_params.is_owner,
+        html_submit_button: $t_html({defaultMessage: "Save changes"}),
+        require_modal_footer: true,
+    });
+
+    $container.append(adminHumanForm);
+
+    let fields_user_pills;
+    (function set_role_dropdown_and_fields_user_pills() {
+        $("#user-role-select").val(user.role);
+        const profileFieldSelector = "#edit-user-form .custom-profile-field-form";
+        settings_account.append_custom_profile_fields(profileFieldSelector, user.user_id);
+        settings_account.initialize_custom_date_type_fields(profileFieldSelector);
+        fields_user_pills = settings_account.initialize_custom_user_type_fields(
+            profileFieldSelector,
+            user.user_id,
+            true,
+            false,
+        );
+    })();
+
+    $("#edit-user-form").on("click", ".dialog_submit_button", () => {
+        const role = Number.parseInt($("#user-role-select").val().trim(), 10);
+        const $full_name = $("#edit-user-form").find("input[name='full_name']");
+        const profile_data = settings_users.get_human_profile_data(fields_user_pills);
+        const url = "/json/users/" + encodeURIComponent(user.user_id);
+        const data = {
+            full_name: $full_name.val(),
+            role: JSON.stringify(role),
+            profile_data: JSON.stringify(profile_data),
+        };
+        const $submitBtn = $("#edit-user-form .dialog_submit_button");
+        const $cancelBtn = $("#edit-user-form .dialog_cancel_button");
+        show_button_spinner($submitBtn);
+        $cancelBtn.prop("disabled", true);
+
+        channel.patch({
+            url,
+            data,
+            success() {
+                hide_user_profile();
+            },
+            error(xhr) {
+                ui_report.error($t_html({defaultMessage: "Failed"}), xhr, $("#dialog_error"));
+                hide_button_spinner($submitBtn);
+                $cancelBtn.prop("disabled", false);
+            },
+        });
+    });
```

Comment:
Is there a copy of this function or other related code in `settings_users.js` that is now unused and should be deleted?  

I suppose the same question applies to `deactivate_user_button`, as well.

Reply:
So both these functions are nested inside the function, `show_edit_user_info_modal`, that renders the "Manage user" standalone modal from before. However, that function is still being called in **Settings -> Manage Organization -> Users -> Edit **. If we switch to using the full profile modal for this; we can remove the function as it isn't being used anywhere else.

Reply:
I think the vision is that we could have that edit button pop this modal as well, since we expect it to be a better experience; perhaps you can try doing that change in a new final commit, so we can back the change out if it isn't working?